src/common/HeartbeatMap.cc

```diff
@@ -90,14 +96,14 @@ void HeartbeatMap::reset_timeout(heartbeat_handle_d *h, time_t grace, time_t sui
 {
   ldout(m_cct, 20) << "reset_timeout '" << h->name << "' grace " << grace
 		   << " suicide " << suicide_grace << dendl;
-  time_t now = time(NULL);
+  auto now = ceph::coarse_mono_clock::now();
   _check(h, "reset_timeout", now);
 
-  h->timeout = now + grace;
+  h->timeout = tp_to_seconds(now) + grace;
```

Comment:
might want to use `std::atomic<ceph::coarse_mono_clock>` for `heartbeat_handle_d::timeout` and `heartbeat_handle_d::suicide_timeout`

Reply:
it looks complicated if we set std::atomic<ceph::coarse_mono_clock>. https://stackoverflow.com/questions/29364456/stdatomicstdchronohigh-resolution-clocktime-point-can-not-compile

Reply:
#17213

Reply:
@XinzeChi sorry, i just figured out that this is a bug in GCC. see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53901 and https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53903.